packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts

```diff
@@ -729,9 +732,24 @@ const fetchGraphql = async ({
       requestOptions.auth = htaccessCredentials
     }
 
-    response = await moduleHelpers
-      .getHttp(limit)
-      .post(url, { query, variables }, requestOptions)
+    response = await retry(
+      (bail: (e: Error) => void) =>
+        moduleHelpers
+          .getHttp(limit)
+          .post(url, { query, variables }, requestOptions)
+          .catch(e => {
+            if (!errorIs500ish(e)) {
+              // for any error that is not a 50x error, we bail, meaning we stop retrying. error will be thrown one level higher
+              bail(e)
+
+              return null
+            } else {
+              // otherwise throwing the error will cause the retry to happen again
+              throw e
+            }
+          }),
+      { retries: 5 }
```

Comment:
Is it worth making this configurable via plugin options?

Reply:
The plugin is super bloated with options so I think we should leave it out for now and add it later if needed. It's very easy to add and hard to take away 😅